### PR TITLE
app-text/highlight: version bump to 3.57

### PR DESCRIPTION
**Pull request by cjmayo**
Signed-off-by: Chris Mayo <aklhfex@gmail.com>
Package-Manager: Portage-2.3.62, Repoman-2.3.11

---

- Bumped to EAPI 7
- qmake-fix.patch included upstream
- now possible to disable compression of man pages and avoid QA warning

```diff
--- highlight-3.42-r1.ebuild
+++ highlight-3.50.ebuild
@@ -1,7 +1,7 @@
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
+EAPI=7
 
 inherit flag-o-matic qmake-utils toolchain-funcs
 
@@ -11,7 +11,7 @@
 
 LICENSE="GPL-3"
 SLOT="0"
-KEYWORDS="alpha amd64 arm arm64 hppa ia64 ppc ppc64 s390 ~sh sparc x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux ~ppc-macos"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux ~ppc-macos"
 IUSE="examples qt5"
 
 RDEPEND="
@@ -42,10 +42,11 @@
 	"conf_dir=${EPREFIX}/etc/highlight/"
 )
 
-PATCHES=( "${FILESDIR}"/${P}-qmake-fix.patch ) # bug 649398
-
 src_prepare() {
 	default
+
+	# disable man page compression
+	sed -e "/GZIP/d" -i makefile || die
 
 	sed -e "/LSB_DOC_DIR/s:doc/${PN}:doc/${PF}:" \
 		-i src/core/datadir.cpp || die
```

### CODE REVIEW

File: app-text/highlight/highlight-3.50.ebuild [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	qt5? (
+		dev-libs/double-conversion:=
+		dev-qt/qtcore:5
+		dev-qt/qtgui:5
+		dev-qt/qtwidgets:5
+	)
+"
+DEPEND="${RDEPEND}
+	dev-libs/boost
+	virtual/pkgconfig
```
Comment (arthurzam) [inline app-text/highlight/highlight-3.50.ebuild]: Should be moved to BDEPEND.

File: app-text/highlight/highlight-3.50.ebuild [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	qt5? ( dev-qt/linguist-tools:5 )
```
Comment (arthurzam) [inline app-text/highlight/highlight-3.50.ebuild]: Should be moved to BDEPEND

File: app-text/highlight/highlight-3.50.ebuild (hunk 1 above)
Comment (cjmayo) [reply app-text/highlight/highlight-3.50.ebuild]: Moved to BDEPEND.

File: app-text/highlight/highlight-3.50.ebuild (hunk 2 above)
Comment (cjmayo) [reply app-text/highlight/highlight-3.50.ebuild]: Moved to BDEPEND.

**Comment by cjmayo** [comment]
As well as adding BDEPEND I've dropped dev-libs/double-conversion from RDEPEND (and added a message to the commit), it appears to be a qtcore dependency.

**Comment by arthurzam** [comment]
@cjmayo I'm not sure `boost` should be in BDEPEND. I think it should be a DEPEND, but need more info and/or checks.

**Comment by cjmayo** [comment]
I believe highlight is using Boost header-only libraries and there is no linking going on.

Looking at the executables produced:
```
$ ldd /usr/bin/highlight | grep boost
$ ldd /usr/bin/highlight-gui | grep boost
```

**Comment by cjmayo** [comment]
Updated to 3.52, also dropped flag-o-matic.eclass can't see any direct use.

**Comment by a17r** [comment]
> I believe highlight is using Boost header-only libraries and there is no linking going on.
That exactly means it has to go to DEPEND instead.

**Comment by cjmayo** [comment]
OK, boost moved to DEPEND.

**Pull request closed without merging**